lib/r10k/puppetfile.rb

```diff
@@ -195,7 +195,7 @@ def visitor_thread(visitor, mods_queue)
       begin
         while mod = mods_queue.pop(true) do mod.accept(visitor) end
       rescue ThreadError => e
-        logger.error _("Thread error during concurrent module deploy: %{message}") % {message: e.message}
+        logger.debug _("Module thread %{id} exiting: %{message}") % {message: e.message, id: Thread.current.object_id}
```

Comment:
After re-reading the docs a bit, it looks like `ThreadError` is raised by `.pop(true)` when the queue is empty. Logging this at DEBUG level seems reasonable. However, we should add an additional rescue clause at the end of the block to catch and log errors that occur during `mod.accept`:

```ruby
rescue => e
  logger.error _("Error during concurrent deploy of module %{mod_name}: %{message}") %
    {mode_name: mod.name,
     message: e.message}
  Thread.exit
end
```

Reply:
Added additional rescue block